src/transforms/throttle.rs

```diff
@@ -121,74 +121,68 @@ where
-                                            }
-                                            _ => {
-                                                if let Some(key) = key {
-                                                  emit!(ThrottleEventDiscarded{key})
-                                                } else {
-                                                  emit!(ThrottleEventDiscarded{key: "None".to_string()})
-                                                }
-                                            }
+        Box::pin(stream! {
+          loop {
+            let done = tokio::select! {
+                biased;
+
+                maybe_event = input_rx.next() => {
+                    match maybe_event {
+                        None => true,
+                        Some(event) => {
+                            let (throttle, event) = match self.exclude.as_ref() {
+                                Some(condition) => {
+                                    let (result, event) = condition.check(event);
+                                    (!result, event)
+                                },
+                                _ => (true, event)
+                            };
+                            let output = if throttle {
+                                let key = self.key_field.as_ref().and_then(|t| {
+                                    t.render_string(&event)
+                                        .map_err(|error| {
+                                            emit!(TemplateRenderingError {
+                                                error,
+                                                field: Some("key_field"),
+                                                drop_event: false,
+                                            })
+                                        })
+                                        .ok()
+                                });
+
+                                match limiter.check_key(&key) {
+                                    Ok(()) => {
+                                        Some(event)
+                                    }
+                                    _ => {
+                                        if let Some(key) = key {
+                                            emit!(ThrottleEventDiscarded{key})
+                                        } else {
+                                            emit!(ThrottleEventDiscarded{key: "None".to_string()})
                                         }
-                                    } else {
-                                        output.push(event)
+                                        None
                                     }
-                                false
+                                }
+                            } else {
+                                Some(event)
+                            };
+                            if let Some(event) = output {
+                                yield event;
```

Comment:
You could drop the assignment and just `yield` directly, but it doesn't make much of a different. Can't `yield` as a sync transform anyway 😄

Reply:
I actually tried that, as it's an obvious cleanup of the conditional, but it changes the generated code such that the result is no longer `Send` (due to the changes around `limiter` IIRC).